qiita_db/support_files/patches/26.sql

```diff
@@ -0,0 +1,25 @@
+-- June 11,, 2015
+-- Adds portal type to analyses
+
+-- Temporarily have default value for non-portaled analyses
+ALTER TABLE qiita.analysis
+ADD portal_type_id bigint NOT NULL DEFAULT 1;
+
+-- Add FK to portal_type_id
+ALTER TABLE qiita.analysis
+ADD CONSTRAINT fk_analysis_portal
+FOREIGN KEY (portal_type_id)
+REFERENCES qiita.portal_type(portal_type_id);
+
+-- Drop defaut value constraint so it is now required to give portal type
+ALTER TABLE qiita.analysis
+ALTER portal_type_id DROP DEFAULT;
+
+-- Remove existing portals and replace with more relevant ones
+UPDATE qiita.study SET portal_type_id = 1 WHERE portal_type_id = 3;
```

Comment:
The qiita portal is implicit, all studies belong to the study portal because all the studies can be seen from Qiita. However, the other case (sloan and hmp2) it is not currently supported. I was unaware that a single study can belong to more than one portal (w/o taking into account Qiita) so this needs more changes in the DB... Good that we figure this out before it got merged!